test/library/draft/arrow/Arrow.chpl

```diff
@@ -0,0 +1,253 @@
+module Arrow {
+    require "ArrowDecl.chpl";
+    public use ArrowDecl;
+    public use SysCTypes, CPtr;
+    // -------------------------- Type Declarations and Functions -------------
+
+    type ArrowArray = GArrowArray;
+
+    proc arrowInt32(): GArrowInt32DataType {
+        return garrow_int32_data_type_new();
+    }
+    
+    proc arrowInt64(): GArrowInt32DataType {
+        return garrow_int64_data_type_new();
+    }
+    proc arrowString(): GArrowStringDataType {
+        return garrow_string_data_type_new();
+    }
+    proc arrowBool(): GArrowBooleanDataType {
+        return garrow_boolean_data_type_new();
+    }
+    //--------------------------- Array building functions --------------------
+
+    proc array(array: [?arrDom] ?arrayType, validity: [?validityDom] int = [-1], 
```

Comment:
Similar with all the functions, but do we need these domains named? The domains don't seem to be used anywhere, so I think we can just get rid of them (e.g., `array: [] ?arrayType, ...`). Also, writing this out, it seems questionable that the function and argument both have the name `array`. Maybe change the argument to have a different name? (arr, a, etc.)

One other thing: there may be a benefit I am not seeing, but I think it would be cleaner to initialize the validity array to just be empty, not `[-1]`. Then we could just check based on size if we want to do anything with the `validity` array